src/moscot/plotting/_utils.py

```diff
@@ -359,45 +360,137 @@ def _plot_temporal(
+    categories: Optional[Union[str, List[str]]] = None,
+    *,
+    push: bool,
+    time_points: Optional[Sequence[K]] = None,
     basis: str = "umap",
-    result_key: str = "plot_tmp",
-    constant_fill_value: float = 0.0,
+    constant_fill_value: float = np.nan,
+    scale: bool = True,
     cont_cmap: Union[str, mcolors.Colormap] = "viridis",
-    title: Optional[str] = None,
+    title: Optional[Union[str, List[str]]] = None,
+    suptitle: Optional[str] = None,
     figsize: Optional[Tuple[float, float]] = None,
     dpi: Optional[int] = None,
+    dot_scale_factor: float = 2.0,
+    na_color: Optional[str] = "#e8ebe9",
     save: Optional[str] = None,
     ax: Optional[Axes] = None,
     show: bool = False,
+    suptitle_fontsize: Optional[float] = None,
     **kwargs: Any,
-) -> mp.figure.Figure:
-    all_keys = adata.obs[temporal_key].unique()
-    if time_points is None:
-        constant_fill_keys: Set[K] = set()
+) -> mpl.figure.Figure:
+
+    fig, axs = plt.subplots(
+        1, 1 if time_points is None else len(time_points), figsize=figsize, dpi=dpi, constrained_layout=True
+    )
+    axs = np.ravel(axs)  # make into iterable
+
+    if not push and time_points is not None:
+        time_points = time_points[::-1]
+
+    if isinstance(title, list):
+        if TYPE_CHECKING:
+            assert isinstance(time_points, list)
+        if len(title) != len(time_points):
+            raise ValueError("If `title` is a list, its length must be equal to the length of `time_points`.")
+        titles = title
     else:
-        constant_fill_keys = set(all_keys) - set(time_points)
-    tmp = np.full(len(adata), np.nan)
-    for t in adata.obs[temporal_key].unique():
-        mask = adata.obs[temporal_key] == t
-        if t in constant_fill_keys:
-            tmp[mask] = constant_fill_value
+        name = "descendants" if push else "ancestors"
+        if time_points is not None:
+            titles = [f"{categories} at time point {start if push else end}"]
+            titles.extend([f"{name} at time point {time_points[i]}" for i in range(1, len(time_points))])
         else:
-            tmp[mask] = adata[adata.obs[temporal_key] == t].obs[key_stored]
-
-    adata.obs[result_key] = tmp
-
-    sc.set_figure_params(figsize=figsize, dpi=dpi)  # TODO(@MUCDK, michalk8): necessary? want to make it uniform
-    fig = sc.pl.embedding(
-        adata=adata,
-        basis=basis,
-        color=result_key,
-        color_map=cont_cmap,
-        title=title,
-        ax=ax,
-        show=show,
-        **kwargs,
-    )
+            titles = [f"{categories} at time point {start if push else end} and {name}"]
+    for i, ax in enumerate(axs):
+        if time_points is None:
+            if scale:
+                vmin, vmax = np.nanmin(adata.obs[key_stored]), np.nanmax(adata.obs[key_stored])
+            else:
+                vmin, vmax = 0, 1
+            adata.obs[f"result_key_{i}"] = (adata.obs[key_stored] - vmin) / (vmax - vmin)
```

Comment:
Why not use `StandardScaler`?

Reply:
Also minor pet peeve: wouldn't introduce additional computations when `scale = False`, as they are not necessary.